### PR TITLE
[Xamarin.Android.Build.Tasks] Fix Aapt2 LowerCasing when using library resources.

### PR DESCRIPTION
**Pull request by dellis1972**
We found an issue where if an application layout file
was referencing a library project resource it was NOT
being lowercased. This would result in the following
error

	error APT0000: resource drawable/Case_Check not found

The reason for this was the fact that we were only passing in
the `$(MonoAndroidResDirIntermediate)` directory to
`ConvertResourcesCases`. The Task was a bit of a strange one in
that it would use the same property for both the list of directories
it needed to alter AND the ones it needed to use for reference.
This puts us in a bit of a pickle because in the case of `aapt2`
we only want to work on that ONE directory.

So this PR introduces an additional parameter to the task

	`AdditionalResourceDirectories`

The purpose of this parameter is to provide a list of directories
to check for resources, but NOT alter or change. They are for
reference only. This will allow use to process a single
directory without being forces to process a bunch of others.
Those "other" directories are already processed by the
`_ConvertLibraryResourcesCases` target. This is part of how
`aapt2` work in that we are creating the `.flata` archives for
each library project as well as a seperate one for the main app.
This allows us to do faster incremental builds since the library
resources should rarely change.

### CODE REVIEW

File: src/Xamarin.Android.Build.Tasks/Tasks/ConvertResourcesCases.cs [hunk 1]
```diff
@@ -83,7 +85,10 @@ void FixupResources (ITaskItem item, Dictionary<string, string> acwMap)
 			}
 			Log.LogDebugMessage ("  AndroidConversionFlagFile modified: {0}", lastUpdate);
 			
-			var resourcedirectories = ResourceDirectories.Where (s => s != item).Select(s => s.ItemSpec).ToArray();
+			var resourcedirectories = ResourceDirectories.Where (s => s != item)
+				.Concat (AdditionalResourceDirectories ?? new ITaskItem [0])
+				.Select(s => s.ItemSpec)
+				.ToArray();
```
[the remark this answers is not in the document]
Comment (dellis1972) [reply src/Xamarin.Android.Build.Tasks/Tasks/ConvertResourcesCases.cs]: no idea. Probably the number of support libs at a minimum.

[the remark this answers is not in the document]
Comment (jonathanpeppers) [reply src/Xamarin.Android.Build.Tasks/Tasks/ConvertResourcesCases.cs]: Maybe it's worth removing the LINQ if we are already changing it? I'm fine though, if you think it's OK.

**Comment by jonpryor** [comment]
Unit test? :-D

Should we migrate over `monodroid/samples/Hello`, as that's where we first saw the issue?  Or have a "code" unit test in `src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests`?

**Comment by dellis1972** [comment]
@jonpryor I will add a unit test in the morning, I ran out of time this evening.

File: src/Xamarin.Android.Build.Tasks/Tasks/ConvertResourcesCases.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (dellis1972) [reply src/Xamarin.Android.Build.Tasks/Tasks/ConvertResourcesCases.cs]: Removed the Link

**Comment by jonpryor** [comment]
Shouldn't `Xamarin.Android-Tests.sln` be updated to include the new `samples/HelloWorld/HelloLibrary/HelloLibrary.csproj` file?

File: Xamarin.Android-Tests.sln [hunk 2]
```diff
@@ -90,128 +92,195 @@ Global
 	GlobalSection(SolutionConfigurationPlatforms) = preSolution
 		Debug|Any CPU = Debug|Any CPU
 		Release|Any CPU = Release|Any CPU
+		Bundle|Any CPU = Bundle|Any CPU
```
Comment (dellis1972) [inline Xamarin.Android-Tests.sln]: @jonpryor I really hate modifying the sln is VSForMac, it does all sorts or weird stuff.

**Pull request merged**